File: Getting Started - WCF RIA Services/[C#]-Getting Started - WCF RIA Services/C#/HRApp/Helpers/StringFormatValueConverter.cs
namespace HRApp
{
    using System;
    using System.Windows.Data;

    /// <summary>
    ///     Two way IValueConverter that lets you bind a property on a bindable object
    ///     that can be an empty string value to a dependency property that should 
    ///     be set to null in that case
    /// </summary>
    public class StringFormatValueConverter : IValueConverter
    {
        private readonly string formatString;

        /// <summary>
        ///     Creates a new <see cref="StringFormatValueConverter"/>
        /// </summary>
        /// <param name="formatString">Format string, it can take zero or one parameters, the first one being replaced by the source value</param>
        public StringFormatValueConverter(string formatString)
        {
            this.formatString = formatString;
        }

        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return string.Format(System.Globalization.CultureInfo.CurrentUICulture, this.formatString, value);
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            throw new NotSupportedException();
        }
    }
}

File: Getting Started - WCF RIA Services/[VB]-Getting Started - WCF RIA Services/VB/HRApp/Helpers/StringFormatValueConverter.vb
Imports System
Imports System.Globalization
Imports System.Windows.Data

''' <summary>
'''     One way IValueConverter that lets you bind a dependency property to
'''     the an object by converting the object's value to a string representation
'''     using <code>String.Format</code>
''' </summary>
Public Class StringFormatValueConverter
    Implements IValueConverter

    Private ReadOnly formatString As String

    ''' <summary>
    '''     Creates a new <see cref="StringFormatValueConverter"/>
    ''' </summary>
    ''' <param name="formatString">Format string, it can take zero or one parameters, the first one being replaced by the source value</param>
    Public Sub New(ByVal formatString As String)
        Me.formatString = formatString
    End Sub

    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
        Return String.Format(CultureInfo.CurrentUICulture, Me.formatString, value)
    End Function

    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
        Throw New NotSupportedException()
    End Function
End Class